lib/awful/hotkeys_popup/widget.lua

```diff
@@ -295,6 +367,9 @@ function widget.new(args)
             end
         end
         self:_sort_hotkeys(self._cached_awful_keys)
+        if self.merge_duplicates then
+            self:_abbreviate_hotkeys(self._cached_awful_keys)
```

Comment:
why it's called from inside `_import_awful_keys()`?

Reply:
I put it after `_sort_hotkeys` because I supposed that's where you start processing the keys. A bad judgment call on my part, I guess.

I was looking for any method that's called before joining the texts, coloring the texts and laying out the columns is fine.

Reply:
> I put it after _sort_hotkeys
i understand the logic, but in that case `_sort_hotkeys` is called from two different places, and with different target for sorting, i suppose what abbreviating should happen in the same way

Reply:
The other call of `_sort_hotkeys` is for cheatsheets for external programs (the files under `lib/awful/hotkeys_popup/keys`).

Those cheatsheets weren't created by `awful.key`, they're hand-made, and they already were abbreviated by hand, too. No need to use `_abbreviate_hotkeys` there.

Reply:
that should be then explicitly mentioned in docs for `.merge_duplicates`